JavaScript/src/utilities.ts

```diff
@@ -70,11 +70,50 @@ export class Match {
 
 export class RegExpUtility {
     static getMatches(regex: RegExp, source: string): Array<Match> {
+        let rawRegex:string = (regex as any).xregexp.source;
```

Comment:
Can you add some comments in the code to explain the implemented logic?
It will help future people when they need to maintain/extend this.
Thanks.